pkg/indexmanagement/reconcile.go

```diff
@@ -0,0 +1,288 @@
+package indexmanagement
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	batchv1 "k8s.io/api/batch/v1"
+	batch "k8s.io/api/batch/v1beta1"
+	core "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apis "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/logger"
+	k8sfactory "github.com/openshift/elasticsearch-operator/pkg/types/k8s/factory"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/openshift/elasticsearch-operator/pkg/utils/comparators"
+)
+
+const (
+	indexManagmentNamePrefix = "ocp-index-mgm"
+	indexManagementConfigmap = "indexmanagement-scripts"
+	defaultShardSize         = int32(40)
+)
+
+var (
+	defaultCPURequest      = resource.MustParse("100m")
+	defaultMemoryRequest   = resource.MustParse("32Mi")
+	jobHistoryLimitFailed  = utils.GetInt32(5)
+	jobHistoryLimitSuccess = utils.GetInt32(5)
+
+	millisPerSecond = uint64(1000)
+	millisPerMinute = uint64(60 * millisPerSecond)
+	millisPerHour   = uint64(millisPerMinute * 60)
+	millisPerDay    = uint64(millisPerHour * 24)
+	millisPerWeek   = uint64(millisPerDay * 7)
+)
+
+type rolloverConditions struct {
+	MaxAge  string `json:"max_age,omitempty"`
+	MaxDocs int32  `json:"max_docs,omitempty"`
+	MaxSize string `json:"max_size,omitempty"`
+}
+
+func ReconcileCurationConfigmap(apiclient client.Client, cluster *apis.Elasticsearch) error {
+	data := map[string]string{
+		"delete": deleteScript,
```

Comment:
where does `deleteScript` come from?

Reply:
its defined in a file called `scripts.go` instead of loading it from a file into the configmap